src/pages/HomePage.js

```diff
@@ -1,130 +1,155 @@
-	}));
-	const getData = async () => {
-		const res = await apiService.get(`/car?page=${page}`);
-		setCars(res.cars);
-		setTotalPages(res.total);
-	};
-	useEffect(() => {
-		getData();
-	}, [page]);
+  const handleClickDelete = (id) => {
+    setOpenConfirm(true);
+    setSelectedCar(cars.find((car) => car._id === id));
+  };
+  const handleDelete = async () => {
+    try {
+      await apiService.delete(`/cars/${selectedCar._id}`);
+      getData();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const name =
+    selectedCar?.release_date +
+    " " +
+    selectedCar?.make +
+    " " +
+    selectedCar?.model;
+  const columns = [
+    { field: "name", headerName: "Name", flex: 3, minWidth: 120 },
+    { field: "style", headerName: "Style", flex: 1, minWidth: 120 },
+    { field: "size", headerName: "Size", flex: 1, minWidth: 100 },
+    {
+      field: "transmission_type",
+      headerName: "Transmission Type",
+      flex: 1.5,
+      minWidth: 120,
+    },
+    { field: "price", headerName: "Price", flex: 1, minWidth: 80 },
+    { field: "release_date", headerName: "Year", flex: 1, minWidth: 80 },
+    {
+      field: "id",
+      headerName: "Edit/Delete",
+      minWidth: 120,
+      flex: 1,
+      sortable: false,
+      renderCell: ({ value }) => (
+        <Stack direction="row" spacing={1}>
+          <IconButton onClick={() => handleClickEdit(value)}>
+            <EditIcon />
+          </IconButton>
+          <IconButton onClick={() => handleClickDelete(value)}>
+            <DeleteIcon />
+          </IconButton>
+        </Stack>
+      ),
+    },
+  ];
+  const rows = cars.map((car) => ({
+    id: car._id,
+    name: car.make + " " + car.model,
+    size: car.size,
+    style: car.style,
+    transmission_type: car.transmission_type,
+    price: car.price,
+    release_date: car.release_date,
+  }));
 
-	return (
-		<Container maxWidth="lg" sx={{ pb: 3 }}>
-			<ConfirmModal
-				open={openConfirm}
-				name={name}
-				handleClose={() => {
-					setOpenConfirm(false);
-				}}
-				action={handleDelete}
-			/>
-			<FormModal
-				open={openForm}
-				refreshData={() => {
-					setOpenForm(false);
-					setSelectedCar(null);
-					getData();
-				}}
-				selectedCar={selectedCar}
-				handleClose={() => {
-					setOpenForm(false);
-					setSelectedCar(null);
-				}}
-				mode={mode}
-			/>
-			<div style={{ height: 630, width: '100%' }}>
-				<DataGrid
-					disableSelectionOnClick
-					rows={rows}
-					rowCount={5 * totalPages}
-					columns={columns}
-					rowsPerPageOptions={[]}
-					components={{
-						Pagination: () => <Pagination color="primary" count={totalPages} page={page} onChange={(e, val) => setPage(val)} />,
-					}}
-				/>
-			</div>
-			<Fab variant="extended" color="info" onClick={handleClickNew} sx={{ position: 'fixed', bottom: 10, left: 10 }} aria-label="add">
-				<AddIcon />
-				New
-			</Fab>
-		</Container>
-	);
-};
+  const getData = useCallback(async () => {
```

Comment:
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

**The recommended fix is to move that function inside of your effect.**

Do something like this: 
![image](https://user-images.githubusercontent.com/63140320/194215062-73634a3a-32cb-4df8-b5fb-8795e91af0b1.png)

`useCallback` hook is a last resort, so let's avoid this.

Reply:
in the existing code, getData() is also called elsewhere outside of the useEffect hook so I can't move the declaration inside useEffect, hence i resorted to useCallback() to avoid the missing dependency warning